File: Aug/Day1/pascals-triangle.js
//  Time: O(N^2)  ::  Space: O(N^2)
class Solution {
    generate(numRows) {
        let nums = [1];
        let pascalList = [nums];

        for (let i = 1; i < numRows; i++) {
            let temp = new Array(nums.length + 1).fill(1);
            for (let j = 1; j < temp.length - 1; j++) {
                temp[j] = nums[j] + nums[j - 1];
            }
            nums = temp;
            pascalList.push(nums);
        }

        return pascalList;
    }
}

File: Aug/Day1/pascals-triangle.py
#  Time: O(N^2)  ::  Space: O(N^2)
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        nums = [1]
        pascalList = [nums]
        for i in range(numRows-1):
            temp =[1]* (len(nums)+1)
            for j in range(1, len(temp)-1) :
                    temp[j]= nums[j]+nums[j-1]
            nums = temp
            pascalList.append(nums)
        return pascalList